source/Preferences.cpp

```diff
@@ -317,14 +318,18 @@ void Preferences::SetScrollSpeed(int speed)
 // View zoom.
 double Preferences::ViewZoom()
 {
-	return ZOOMS[zoomIndex];
+	const auto &zooms = GameData::Interfaces().Get("hud")->GetList("zooms");
+	if(zoomIndex >= zooms.size())
+		return zooms.back();
```

Comment:
You need to ensure the vector isn't empty.
Calling `back` on an empty vector is undefined.
https://en.cppreference.com/w/cpp/container/vector/back